### ISSUE TITLE
Improve Site Accessibility

### ISSUE THREAD

**Issue by cassandralfernandes** [civiform/civiform#3243]
### Epic Overview
Building on top of our existing standards, this work will exceed requirements to create an excellent UX that is accessible to all users.

### Problem Statements(s)
The majority of the bugs found were related to:

- Screen reader (label, focus, unable to reach controls), keyboard focus, improper headings (No H1's on page, heading hierarchy), error cases, color contrast, Large text , high contrast, large text, page titles etc.
- Missing/Improper labels prevent screen reader users from using the site as they would be unable to identify the purpose of the form fields.
- Section headings within pages allows screen reader users to jump sections within pages instead of moving onto every control on page
- Color contrast issues would make it difficult for individuals with visual impairments from seeing certain items on the page hence, preventing them from using the site.

### Design & Eng Requirements
[See all accessibility issues here](https://github.com/civiform/civiform/labels/a11y) 

### EOQ update 
In Q3, the following a11y progress was made:
- Added automated a11y testing
- Conducted UXR with a screen reader user
- Conducted a full accessibility audit against WCAG 2.1 (Web Content Accessibility Guidelines)
- A11y features and bug fixes including but not limited to:
        - Read out question errors and descriptions for screenreader users
        - Improved consistency between visual and screenreader text
        - Removed duplicated screenreader only headings, improved logical heading groupings, added more heading indicators
        - Restructured HTML question rendering to use screenreader compatible grouping
        - Mark visual elements appropriately either as should-be-read or should-be-ignored by screenreaders
        
In Q4, we will work towards exceeding WCAG 2.1 a11y standards. See #3599 to track progress on Q4 a11y goals
